kornia/augmentation/_2d/geometric/center_crop.py

```diff
@@ -110,14 +110,17 @@ def apply_transform(
         if self.flags["cropping_mode"] == "slice":  # uses advanced slicing to crop
             # TODO: implement as separated function `crop_and_resize_iterative`
             B, C, _, _ = input.shape
-            H, W = self.size
-            out = torch.empty(B, C, H, W, device=input.device, dtype=input.dtype)
+            x1 = params["src"].to(device="cpu", dtype=torch.long)[:, 0, 0].numpy()
```

Comment:
It's not boilerplate to try to cast and send to device four times? Since it's a single tensor, it can be done in a single call